scripts/ci-entrypoint.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+
+# To run locally, set GCP_CLIENT_ID, GCP_CLIENT_SECRET, GCP_SUBSCRIPTION_ID, GCP_TENANT_ID
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+cd "${REPO_ROOT}" || exit 1
+
+# shellcheck source=hack/ensure-go.sh
+source "${REPO_ROOT}/hack/ensure-go.sh"
+# shellcheck source=hack/ensure-kind.sh
+source "${REPO_ROOT}/hack/ensure-kind.sh"
+
+# building kubectl from tools folder
+mkdir -p "${REPO_ROOT}/hack/tools/bin"
+KUBECTL=$(realpath hack/tools/bin/kubectl)
+# export the variable so it is available in bash -c wait_for_nodes below
+export KUBECTL
+make "${KUBECTL}" &>/dev/null
+echo "KUBECTL is set to ${KUBECTL}"
+
+# shellcheck source=hack/ensure-kustomize.sh
+source "${REPO_ROOT}/hack/ensure-kustomize.sh"
+# shellcheck source=hack/ensure-tags.sh
+source "${REPO_ROOT}/hack/ensure-tags.sh"
+# shellcheck source=hack/parse-prow-creds.sh
+source "${REPO_ROOT}/hack/parse-prow-creds.sh"
+# shellcheck source=hack/util.sh
+source "${REPO_ROOT}/hack/util.sh"
+
+get_random_region=(){
+    local REGIONS=("asia-east1-a" "asia-east2-b" "asia-northeast3-c" "asia-south1-a" "europe-north1-a" "europe-west3-c" "northamerica-northeast2-b")
+    echo "${REGIONS[${RANDOM} % ${#REGIONS[@]}]}"
+}
+
+setup(){
+	# setup REGISTRY for custom images.
+    : "${REGISTRY:?Environment variable empty or not defined.}"
+    if [[ -z "${CLUSTER_TEMPLATE:-}" ]]; then
+        select_cluster_template
+    fi
+
+    export GCP_REGION="<GCP_REGION>"
+    export GCP_PROJECT="<GCP_PROJECT>"
+    export KUBERNETES_VERSION=1.20.9
+    export GCP_CONTROL_PLANE_MACHINE_TYPE=n1-standard-2
+    export GCP_NODE_MACHINE_TYPE=n1-standard-2
+    export GCP_NETWORK_NAME=<GCP_NETWORK_NAME or default>
+    export CLUSTER_NAME="<CLUSTER_NAME>"
+}
+
+wait_for_nodes() {
```

Comment:
this is not needed

Reply:
@cpanato from which line number shall I check that these are not needed?